src/lifecycle/created.js

```diff
@@ -1,93 +1,59 @@
-import assignSafe from '../util/assign-safe';
+import apiProperty from '../api/property';
+import createdOnDescendants from './created-on-descendants';
 import data from '../util/data';
 import events from './events';
-import properties from './properties';
-import protos from '../util/protos';
-import registry from '../global/registry';
-import render from './render';
-import walkTree from '../util/walk-tree';
-
-let elProto = window.Element.prototype;
-let oldSetAttribute = elProto.setAttribute;
-let oldRemoveAttribute = elProto.removeAttribute;
-
-function applyPrototype (proto) {
-  let prototypes = protos(proto);
-  return function () {
-    prototypes.forEach(proto => {
-      if (!proto.isPrototypeOf(this)) {
-        assignSafe(this, proto);
-      }
-    });
-  };
+import patchAttributeMethods from './patch-attribute-methods';
+import propertiesCreated from './properties-created';
+import propertiesReady from './properties-ready';
+import prototype from './prototype';
+import renderer from './renderer';
+import resolve from './resolve';
+
+// TODO Remove this when we no longer support the legacy definitions and only
+// support a superset of a native property definition.
+function ensurePropertyFunctions (opts) {
+  let props = opts.properties;
+  let names = Object.keys(props || {});
+  return names.reduce(function (prev, curr) {
+    prev[curr] = opts.properties[curr];
+    if (typeof prev[curr] !== 'function') {
+      prev[curr] = apiProperty(prev[curr]);
+    }
+    return prev;
+  }, {});
```

Comment:
`prev` and `curr` as parameter names are confusing here: why not just `obj`/`descriptor` and `name`?